src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java

```diff
@@ -50,6 +50,31 @@
  * &lt;module name=&quot;ParameterAssignment&quot;/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class MyClass {
+ *   int method(int parameter) {
+ *     if (parameter &lt;= 0 ) {
+ *       throw new IllegalArgumentException("A positive value is expected");
+ *     }
+ *     parameter -= 2;  // violation
+ *     return parameter;
+ *   }
+ * }
+ *
+ * class MyClass {
+ *   int method(int parameter) {
+ *     int var = parameter;
```

Comment:
Please
1. rename variable (var is not a good name)
2. move it after if block (check original parameter in if condition)